dht.go

```diff
@@ -193,31 +219,45 @@ func makeDHT(ctx context.Context, h host.Host, dstore ds.Batching, protocols []p
 		cmgr.UntagPeer(p, "kbucket")
 	}
 
+	return rt, err
+}
+
+func makeDHT(ctx context.Context, h host.Host, cfg *opts.Options) (*IpfsDHT, error) {
+	rt, err := makeRoutingTable(h, cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to construct routing table,err=%s", err)
+	}
+
 	dht := &IpfsDHT{
-		datastore:        dstore,
-		self:             h.ID(),
-		peerstore:        h.Peerstore(),
-		host:             h,
-		strmap:           make(map[peer.ID]*messageSender),
-		ctx:              ctx,
-		providers:        providers.NewProviderManager(ctx, h.ID(), dstore),
-		birth:            time.Now(),
-		routingTable:     rt,
-		protocols:        protocols,
-		bucketSize:       bucketSize,
-		triggerRtRefresh: make(chan chan<- error),
-	}
-
-	var err error
+		datastore:         cfg.Datastore,
+		self:              h.ID(),
+		peerstore:         h.Peerstore(),
+		host:              h,
+		strmap:            make(map[peer.ID]*messageSender),
+		ctx:               ctx,
+		providers:         providers.NewProviderManager(ctx, h.ID(), cfg.Datastore),
+		birth:             time.Now(),
+		routingTable:      rt,
+		protocols:         cfg.Protocols,
+		bucketSize:        cfg.BucketSize,
+		triggerRtRefresh:  make(chan chan<- error),
+		triggerSelfLookup: make(chan chan<- error),
+	}
+
 	evts := []interface{}{&event.EvtPeerIdentificationCompleted{}, &event.EvtPeerIdentificationFailed{}}
 	dht.subscriptions.evtPeerIdentification, err = h.EventBus().Subscribe(evts, eventbus.BufSize(256))
 	if err != nil {
 		logger.Errorf("dht not subscribed to peer identification events; things will fail; err: %s", err)
 	}
 
+	dht.subscriptions.everPeerAddressChanged, err = h.EventBus().Subscribe(&event.EvtLocalAddressesUpdated{}, eventbus.BufSize(256))
+	if err != nil {
+		logger.Errorf("dht not sybscribed to peer address changed event; err=%s ", err)
```

Comment:
```suggestion
		logger.Errorf("dht not subscribed to peer address changed event; err=%s ", err)
```

Reply:
@willscott Oh that's just how we spell it down here. It's a very us thing.